src/ui/public/courier/search_source/search_source.js

```diff
@@ -84,14 +84,35 @@ import { FieldWildcardProvider } from '../../field_wildcard';
 import { getHighlightRequest } from '../../../../core_plugins/kibana/common/highlight';
 import { BuildESQueryProvider } from './build_query';
 
-function parseInitialState(initialState) {
-  if (!initialState) {
+const SETTABLE_PROPS = [
+  'type',
+  'query',
+  'filter',
+  'sort',
+  'highlight',
+  'highlightAll',
+  'aggs',
+  'from',
+  'searchAfter',
+  'size',
+  'source',
+  'version',
+  'fields',
+];
+
+const GETTABLE_PROPS = [
+  ...SETTABLE_PROPS,
+  'index', // This is settable via `setIndexPattern`
+];
+
+function parseInitialData(initialData) {
```

Comment:
I personally think that "data" is too ambiguous. While `getData()` is better than `toJSON()`, I think `getState()`, `getAllValues()`, or something similar is more of an improvement. "data" can really just be anything, at least "state" is clearly linked to the state of the get/set's that this object supports. "values" mirrors the naming of the `getValue()` and `setValue()` methods. But the "data" could literally be anything IMO.